plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/image/deprecated.tsx

```diff
@@ -1,66 +1,51 @@
 /**
  * External dependencies
  */
-import clsx from 'clsx';
 import { createBlock } from '@wordpress/blocks';
 
 /**
  * Internal dependencies
  */
 import metadata from './block.json';
 import { BlockAttributesV1 } from './types';
+import save from '../save';
 
-// In v2, we removed the `showSaleBadge` attribute and converted it to an inner block.
+// In v2, we're migrating the `showSaleBadge` attribute to an inner block.
 const v1 = {
-	attributes: {
-		...metadata.attributes,
-		showSaleBadge: {
-			type: 'boolean',
-			default: true,
-		},
-		saleBadgeAlign: {
-			type: 'string',
-			default: 'right',
-		},
+	save,
+	attributes: metadata.attributes,
+	isEligible: ( { showSaleBadge }: BlockAttributesV1 ) => {
+		// If the block is pristine, it doesn't have a showSaleBadge attribute
+		// but it is `true` by default.
+		const trueByDefault = showSaleBadge === undefined;
+
+		// If the block was edited, it will have a showSaleBadge attribute
+		// that we should respect.
+		const trueAfterEdit = showSaleBadge === true;
+
+		return trueByDefault || trueAfterEdit;
 	},
-	save( { attributes }: { attributes: BlockAttributesV1 } ) {
-		if (
-			attributes.isDescendentOfQueryLoop ||
-			attributes.isDescendentOfSingleProductBlock
-		) {
-			return null;
-		}
-
-		return <div className={ clsx( 'is-loading', attributes.className ) } />;
-	},
-	isEligible: ( attributes: BlockAttributesV1 ) =>
-		attributes.showSaleBadge !== undefined,
 	migrate: ( attributes: BlockAttributesV1 ) => {
-		const { showSaleBadge, saleBadgeAlign, ...rest } = attributes;
-		if ( ! showSaleBadge ) {
-			return [ rest ];
-		}
+		const { showSaleBadge, saleBadgeAlign } = attributes;
 
-		let justifyContent = 'flex-end';
-		if ( saleBadgeAlign === 'left' ) {
-			justifyContent = 'flex-start';
-		} else if ( saleBadgeAlign === 'center' ) {
-			justifyContent = 'center';
```

Comment:
Mmm, I guess this is actually a problem I hadn't thought about. We are not keeping backwards compatibility with the `saleBadgeAlign` on migration since we migrated to block align. It's a small problem, the users can fix it immediately, but still… not the ideal experience.

Reply:
We do, [I pass it here](https://github.com/woocommerce/woocommerce/pull/57889/files#diff-69e1c96271f9381541b4f08cab28247de165de7ef03c8d7cfc7064910b5b1a92R45-R48) but to the On Sale Badge directly (`align` -> `align`).